src/spec/junit_formatter.cr

```diff
@@ -35,6 +35,10 @@ module Spec
     end
 
     def self.file(output_path : Path)
+      if output_path.extension != ".xml"
+        output_path = output_path.join("output.xml")
+      end
+
```

Comment:
That belongs to `src/spec.cr`, not here.

https://github.com/crystal-lang/crystal/blob/42a64909fae022863e68651a3d35f3c45884efc8/src/spec.cr#L132-L135

Reply:
IMHO `.file` is expecting a path to... the file - not some arbitrary path to mutate afterwards.

Reply:
For me is a path configuration of the output formatter. If there is splitting how to split and which file to create would belong in the formatter and not spec itself since it is not something shared with other formatters like DotFormatter.

Reply:
> [...] since it is not something shared with other formatters like DotFormatter.

That's exactly my point, formatter itself should be responsible for doing any magic with the given path. If I'm calling `Spec::JUnitFormatter.file(path)` i expect to pass a **file path** to it...